crates/enarx-config/src/lib.rs

```diff
@@ -45,30 +55,52 @@ impl<'de> Deserialize<'de> for FileName {
         let name = String::deserialize(deserializer)?;
 
         if name.contains(':') {
-            return Err(D::Error::custom("invalid name"));
+            return Err(D::Error::custom("invalid value for `name` contains ':'"));
         }
 
         Ok(Self(name))
     }
 }
 
+/// The configuration for an Enarx WASI application
+///
+/// This struct can be used with any serde deserializer.
+///
+/// # Examples
+///
+/// ```
+/// extern crate toml;
+/// use enarx_config::EnarxConfig;
+/// const CONFIG: &str = r#"
+/// [[files]]
+/// name = "LISTEN"
+/// kind = "listen"
+/// prot = "tls"
+/// port = 12345
+/// "#;
+///
+/// let config: EnarxConfig = toml::from_str(CONFIG).unwrap();
+/// ```
 #[derive(Clone, Debug, PartialEq, Eq, Deserialize)]
-pub struct Config {
+pub struct EnarxConfig {
+    /// The environment variables to provide to the application
     #[serde(default)]
     pub env: HashMap<String, String>,
 
+    /// The arguments to provide to the application
     #[serde(default)]
     pub args: Vec<String>,
 
+    /// The array of pre-opened file descriptors
     #[serde(default)]
     pub files: Vec<File>,
 
+    /// An optional Steward URL
     #[serde(default)]
-    /// Optional Steward URL
     pub steward: Option<Url>,
 }
 
-impl Default for Config {
+impl Default for EnarxConfig {
```

Comment:
Why the `Enarx` prefix? We're already in `enarx-config` crate, so it could not be any other config.

Reply:
fixed